debian/cloud-init.postinst

```diff
@@ -208,6 +208,16 @@ cleanup_lp1552999() {
         "$edir/cloud-init-local.service" "$edir/cloud-init.service"
 }
 
+remove_deprecated_cloud_config_modules() {
```

Comment:
My only concern with this is that if someone has restructured their module list using square bracket yaml syntax, the '/emit_upstart/d' line could be an issue because it would remove other modules from the config as well.

Reply:
Good suggestion @holmanb I force-push updated the postinst here to add sed magic to redact emit_upstart from a single-line yaml array if present. This is probably an unlikely modification, but I think we can handle it.

Reply:
Manipulating yaml using `sed` feels weird, but I don't have better suggestions at the moment.

Currently the regex assumes that `emit_upstart` won't be last.

```
arc~ echo "[one, two, four, emit_upstart]" | sed -E 's/(.+)(\s+)?emit_upstart(\s+)?,(.+)/\1\4/'                     
[one, two, four, emit_upstart]
arc~ echo "[one, two, four, emit_upstart, five]" | sed -E 's/(.+)(\s+)?emit_upstart(\s+)?,(.+)/\1\4/'
[one, two, four,  five]
```

I _think_ that making the comma match optional is enough to fix this:
```
arc~ echo "[one, two, four, emit_upstart]" | sed -E 's/(.+)(\s+)?emit_upstart(\s+)?,?(.+)/\1\4/'
[one, two, four, ]
```

This leaves a trailing comma, but I _think_ this is fine. Python's yaml module parses it:
```
arc~ echo "[one, two, four, emit_upstart, ]" | sed -E 's/(.+)(\s+)?emit_upstart(\s+)?,?(.+)/\1\4/' |python -c 'import yaml, sys; print(yaml.safe_load(sys.stdin.read()))'
['one', 'two', 'four']
```